source/routing/raptor_api.cpp

```diff
@@ -49,6 +49,129 @@ www.navitia.io
 
 namespace navitia { namespace routing {
 
+// Usefull to protect raptor loop
+static const uint max_nb_raptor_call = 100;
+
+/**
+ * @brief internal function to call raptor in a loop
+ */
+std::vector<Path>
+call_raptor(navitia::PbCreator& pb_creator,
+            RAPTOR& raptor,
+            const map_stop_point_duration& departures,
+            const map_stop_point_duration& destinations,
+            const std::vector<bt::ptime>& datetimes,
+            const type::RTLevel rt_level,
+            const navitia::time_duration& transfer_penalty,
+            const type::AccessibiliteParams& accessibilite_params,
+            const std::vector<std::string>& forbidden_uri,
+            const std::vector<std::string>& allowed_ids,
+            const bool clockwise,
+            const boost::optional<navitia::time_duration>& direct_path_duration,
+            const uint32_t min_nb_journeys,
+            const uint32_t nb_direct_path,
+            const uint32_t max_duration,
+            const uint32_t max_transfers,
+            const size_t max_extra_second_pass,
+            const double night_bus_filter_max_factor,
+            const int32_t night_bus_filter_base_factor)
+{
+    log4cplus::Logger logger = log4cplus::Logger::getInstance(LOG4CPLUS_TEXT("logger"));
+    std::vector<Path> pathes;
+
+    // For each date time
+    DateTime bound = clockwise ? DateTimeUtils::inf : DateTimeUtils::min;
+    for(const auto & datetime : datetimes) {
```

Comment:
minor: Can you just add a TODO on this line telling that it's a deprecated feature and the loop is to be removed? thx

Reply:
we can do that later...